scripts/generate_trials.py

```diff
@@ -0,0 +1,106 @@
+parser.add_argument("--ncpus", "-nc", help="number of CPUs to use", type=int, default=1)
+parser.add_argument("--rss_seed", "-rs", help="random_number_seed", type=int, default=5)
+parser.add_argument("--gseed", "-gs", help="gamma seed for minimizer", type=float, default=3.0)
+
+args=parser.parse_args()
+print(args)
+
+# DO NOT USE PHOTOSPLINE
+skyllh.core.pdf.PHOTOSPLINE_LOADED = False
+
+CFG.from_yaml(args.config)
+
+# Define datasets (todo: generalize to other analyses)
+bp = CFG["paths"]["base_path"]
+dsc_name = CFG["analysis"]["name"]
+dsc = data_samples[dsc_name].create_dataset_collection(base_path=bp)
+ds_names = CFG["analysis"]["datasets"]
+datasets = dsc.get_datasets(ds_names)
+
+# Define the point source at TXS position.
+src_ra  = np.radians(CFG["analysis"]["default_source"]["ra"])
+src_dec = np.radians(CFG["analysis"]["default_source"]["dec"])
+source = PointLikeSource(src_ra, src_dec)
+
+tl = TimeLord()
+
+# Create analysis
+yr=365.25 # days per year
+livetime = CFG["analysis"]["livetime"] * yr
+
+minimizers = {"COBYLA": ScipyMinimizerImpl(method="COBYLA"),
+              "LBFGSB": LBFGSMinimizerImpl()}
+
+
+for min_name, minimizer in minimizers.items():
+    with tl.task_timer('Creating analysis.') as tt:
+        ana = create_analysis(
+            datasets,
+            source,
+            bkg_event_rate_field_names=['astro', 'conv'],
+            refplflux_gamma=args.gamma,
+            gamma_seed = args.gseed,
+            fit_gamma=True,
+            livetime_list=[livetime],
+            compress_data=True,
+            minimizer_impl=minimizer,
+            tl=tl)
+
+    print(tl)
+
+    # Define a random state service.
+    rss_seed = args.rss_seed
+    rss = RandomStateService(rss_seed)
+
+    trials_outfile = os.path.join(CFG["paths"]["output_dir"],
+                                  dsc_name+'_atTXS'+'_gamma{:.2f}_mean_ns{}_gseed{:.2f}_ltime{:.0f}yr_rss{:}_mini{}'.format(args.gamma,
+                                                                                                                            args.mean_ns,
+                                                                                                                            args.gseed,
+                                                                                                                            livetime,
+                                                                                                                            args.rss_seed,
+                                                                                                                            min_name))
+
+    # Run trials.
+    with tl.task_timer('Running trials.') as tt:
+        (seed,mean_ns,mean_ns_null,trial_data)=create_trial_data_file(
+            ana=ana,
+            ncpu=args.ncpus,
+            rss=rss,
+            pathfilename=trials_outfile,
+            n_trials=args.n_trials,
+            mean_n_sig=args.mean_ns,
+            tl=tl)
+
+    print(tl)
```

Comment:
this entire script should be moved to skyllh.examples (potentially file name use_config.py)
script currently uses features that are not on master yet.